tools/oesign/oedump.c

```diff
@@ -179,7 +180,7 @@ void dump_enclave_properties(const oe_sgx_enclave_properties_t* props)
     bool debug = props->config.attributes & OE_SGX_FLAGS_DEBUG;
     printf("debug=%u\n", debug);
 
-    printf("xfrm=%lx\n", props->config.xfrm);
+    printf("xfrm=%" PRIx64 "\n", props->config.xfrm);
 
     printf(
         "num_heap_pages=%llu\n",
```

Comment:
shouldn't this use PRIu64 instead of %llu?

Reply:
In this case, it is wrapped in the macro `OE_LLU` which does some weird cast that I don't quite understand (see include/openenclave/internal/types.h) and the warning doesn't get triggered.

In general it is better to just use PRIu64 and friends. But in the interest of keeping this PR focused I think its best to leave anything that doesn't trigger a warning.

Reply:
Ah I see, that's fine.  On GNUC compilers, uint64_t is "unsigned long" and on MSVC it's "unsigned long long" (both defined in include/openenclave/bits/types.h).   So OE_LLU casts the unsigned long value to unsigned long long for GNUC and doesn't need anything for MSVC.   Using %llu allows the OE_LLU trick whereas PRIx64 requires no cast but instead uses %llu on MSVC and %I64u on GNUC to make the type always match.